### PR TITLE
Decouple Dockerfile into build and train

### PR DESCRIPTION
**Pull request by lissyx**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+DEEPSPEECH_REPO ?= https://github.com/mozilla/DeepSpeech.git
```
Comment (reuben) [inline Makefile:1]: Should we move these docker related files to a "docker" subfolder instead of leaving them in the root?

File: doc/Scorer.rst [hunk 2]
```diff
@@ -24,7 +24,7 @@ Then use the ``generate_lm.py`` script to generate ``lm.binary`` and ``vocab-500
 
 As input you can use a plain text (e.g. ``file.txt``) or gzipped (e.g. ``file.txt.gz``) text file with one sentence in each line.
 
-If you are using a container created from the Dockerfile, you can use ``--kenlm_bins /DeepSpeech/native_client/kenlm/build/bin/``.
+If you are using a container created from the ``Dockerfile.build``, you can use ``--kenlm_bins /DeepSpeech/native_client/kenlm/build/bin/``.
```
Comment (reuben) [inline doc/Scorer.rst:27]:
```suggestion
If you are using a container created from ``Dockerfile.build``, you can use ``--kenlm_bins /DeepSpeech/native_client/kenlm/build/bin/``.
```

File: doc/TRAINING.rst [hunk 3]
```diff
@@ -76,6 +76,22 @@ It has been reported for some people failure at training:
 
 Setting the ``TF_FORCE_GPU_ALLOW_GROWTH`` environment variable to ``true`` seems to help in such cases. This could also be due to an incorrect version of libcudnn. Double check your versions with the :ref:`TensorFlow 1.15 documentation <cuda-deps>`.
 
+Basic Dockerfile for training
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We provide a ``Dockerfile.train`` docker to wrap basic training. You need to generate the dockerfile from the template using:
+This should ensure you to re-use upstream Python3 TensorFlow GPU-enable docker image.
```
Comment (reuben) [inline doc/TRAINING.rst:83]:
```suggestion
We provide ``Dockerfile.train`` to automatically set up a basic training environment in Docker. You need to generate the Dockerfile from the template using:
This should ensure that you'll re-use the upstream Python 3 TensorFlow GPU-enabled Docker image.
```

File: doc/USING.rst [hunk 4]
```diff
@@ -186,6 +186,22 @@ Installing bindings from source
 
 If pre-built binaries aren't available for your system, you'll need to install them from scratch. Follow the :github:`native client build and installation instructions <native_client/README.rst>`.
 
+Dockerfile for building from source
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We provide a ``Dockerfile.build`` docker to wrap rebuilding ``libdeepspeech.so`` as well as C++ ``deepspeech``, Python and others bindings.
+You need to generate the dockerfile from the template using:
```
Comment (reuben) [inline doc/USING.rst:193]:
```suggestion
We provide ``Dockerfile.build`` to automatically build ``libdeepspeech.so``, the C++ native client, Python bindings, and KenLM.
You need to generate the Dockerfile from the template using:
```

File: Makefile (hunk 1 above)
Comment (lissyx) [reply Makefile:1]: I'm not sure, is this really adding noise right now?

Comment (reuben) [reply Makefile:1]: It's not adding noise, just thinking that now we have a top level Makefile that only does a very specific thing.

**Pull request merged**